### PR TITLE
API: Return primary language and repository language stats API URL

### LINKED ISSUE #18395: Add a top language(s) field to `Repository`
### Feature Description

Currently `GET /repos/{owner}/{repo}/languages` is the only way to access this information. However, I am already calling `GET /users/{username}/repos` and would like to avoid the extra request if possible. Don't need the full statistics, just name(s) would suffice.

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by lafriks**
Adds two new properties for Repository structure `langauge` and `languages_url` (named same as in GitHub API)

Fixes #18395

### CODE REVIEW

File: models/repo/repo.go [hunk 1]
```diff
@@ -222,6 +222,30 @@ func (repo *Repository) MustOwner() *user_model.User {
...
+	if err := repo.GetOwner(ctx); err != nil {
+		return fmt.Errorf("load owner: %w", err)
+	}
+
+	// Load primary language
+	stats := make(LanguageStatList, 0, 1)
+	if err := db.GetEngine(ctx).
+		Where("`repo_id` = ? AND `is_primary` = ? AND `language` != ?", repo.ID, true, "other").
+		Find(&stats); err != nil {
+		return fmt.Errorf("find primary languages: %w", err)
```
Comment (Gusted) [inline models/repo/repo.go:237]:
```suggestion
		return fmt.Errorf("couldn't find primary languages: %w", err)
```

File: models/repo/repo.go [hunk 2]
```diff
@@ -222,6 +222,30 @@ func (repo *Repository) MustOwner() *user_model.User {
 	return repo.mustOwner(db.DefaultContext)
 }
 
+// LoadAttributes loads attributes of the repository.
+func (repo *Repository) LoadAttributes(ctx context.Context) error {
+	// Load owner
+	if err := repo.GetOwner(ctx); err != nil {
+		return fmt.Errorf("load owner: %w", err)
```
Comment (Gusted) [inline models/repo/repo.go:229]:
```suggestion
		return fmt.Errorf("couldn't load owner: %w", err)
```

File: models/repo/repo.go [hunk 3]
```diff
@@ -222,6 +222,30 @@ func (repo *Repository) MustOwner() *user_model.User {
...
+	}
+
+	// Load primary language
+	stats := make(LanguageStatList, 0, 1)
```
Comment (Gusted) [inline models/repo/repo.go:233]: Is it possible to avoid using a slice? Given it's only 1 entry.

File: models/repo/repo.go (hunk 1 above)
Comment (lafriks) [reply models/repo/repo.go:237]: it just wraps real error and should not be that verbose, also it does use same wording as in RepositoryList.LoadAttributes

File: models/repo/repo.go (hunk 2 above)
Comment (lafriks) [reply models/repo/repo.go:229]: same as above

File: models/repo/repo.go (hunk 1 above)
Comment (Gusted) [reply models/repo/repo.go:237]: Why wrap then?

File: models/repo/repo.go (hunk 3 above)
Comment (lafriks) [reply models/repo/repo.go:233]: That would require introducing duplicate code for `LanguageStat` attributes with no real gain. (`LanguageStatList` has `LoadAttributes` while `LanguageStat` does not have)

File: models/repo/repo.go (hunk 1 above)
Comment (lafriks) [reply models/repo/repo.go:237]: so that error chain can be easier to trace

**Review by Gusted — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Comment by lafriks** [comment]
🚀

**Pull request merged**